examples/medplum-demo-bots/src/hl7-integration/setup-medplum-agent.ts

```diff
@@ -114,11 +114,11 @@ export async function handler(medplum: MedplumClient, event: BotEvent<Questionna
       const ip = answers['device-address']?.at(index)?.valueString;
       const port = answers['device-port']?.at(index)?.valueInteger;
       const deviceId = `${toIdentifier(organizationName)}-${toIdentifier(deviceName.valueString ?? '')}`;
-      return medplum.upsertResource(
+      return medplum.upsertResource<Device>(
         {
           resourceType: 'Device',
           identifier: [{ system: 'http://example.com/device-id', value: deviceId }],
-          name: deviceName.valueString,
+          deviceName: [{ name: deviceName.valueString ?? '', type: 'user-friendly-name' }],
```

Comment:
nit: The `device-name` should not be a empty entry here. Does the Questionnaire have input validation that would ensure a value is provided?